scylla/src/transport/topology.rs

```diff
@@ -184,6 +184,12 @@ enum PreCqlType {
         type_: PreCollectionType,
     },
     Tuple(Vec<PreCqlType>),
+    Vector {
+        type_: Box<PreCqlType>,
+        /// as per <https://cassandra.apache.org/doc/latest/cassandra/reference/vector-data-type.html>
+        /// vectors are limited to a size of 8192
+        size: u16,
```

Comment:
Now I can see that this is the size limit of a Vector, but what interests me is how Cassandra represents Vector in the system tables. It surely does not use unsigned int, because AFAIK unsigned numbers are not supported by neither ScyllaDB nor Cassandra.
The idea of `PreCqlType` in `topology.rs` is to represent the raw type definition read from system tables. Only later can we convert it to more convenient representation, involving unsigned ints.

Reply:
I believe it makes more sense to store size in the same type as Cassandra does.

Reply:
The java driver uses an int, so I'll do the same: https://github.com/apache/cassandra-java-driver/blob/85bb4065098b887d2dda26eb14423ce4fc687045/core/src/main/java/com/datastax/oss/driver/api/core/type/DataTypes.java#L77

Reply:
I adjusted the code to use an i32 to match the java driver.
I also renamed the `size` to `dimensions` as that is a more accurate name.

Reply:
Unresolving this comment because we talked about that with @wprzytula 
I'm not aware of any case were size of vector is stored in some table (as a cql type). IIUC we always parse type names from strings - so the argument about system tables doesn't apply.
Is it possible to create vector with negative length? I'd be very surprised if it was possible. 
If it is not we can and should use unsigned integer for the size (and simplify the parsing function thanks to that).

Should it be u16? I'm not sure. If documentation says that size is limited to 8192 then yes, but I can't find this information is the link in the comment. Could you link the source?

Reply:
The limit of 2^13 = 8192 is taken from https://cassandra.apache.org/doc/latest/cassandra/reference/vector-data-type.html

However it can be observed that the documentation is incorrect.

From functionally testing, the highest value accepted when creating a table is `2**31 -1 = 2147483647`.
The reason would be that its the highest value representable by a signed 32 bit integer
From functional testing it can also be seen that negative numbers and `0` are rejected.

Do you want me to swap the value from an i32 to a u32?